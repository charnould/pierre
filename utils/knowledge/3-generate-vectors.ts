import { Database } from 'bun:sqlite'
import { openai } from '@ai-sdk/openai'
import { embed } from 'ai'
import * as sqliteVec from 'sqlite-vec'

Database.setCustomSQLite('/opt/homebrew/opt/sqlite/lib/libsqlite3.dylib') // when coding on Mac OS X
const db = new Database('./utils/knowledge/datastore.sqlite')
sqliteVec.load(db)

console.log('ðŸ‘‰ Start building Knowledge database')
console.log('   Estimated duration: 10-15 min')
console.log('   DO *NOT* CLOSE YOUR SHELL!')

db.exec('CREATE VIRTUAL TABLE vectors USING vec0(vector float[3072])')
const query = db.query('SELECT rowid, * FROM chunks').all()

for await (const q of query) {
  const { embedding: vector } = await embed({
    model: openai.embedding('text-embedding-3-large'),
    value: q.chunk
  })

  db.prepare('INSERT INTO vectors(rowid, vector) VALUES (?, vec_f32(?))').run(
    q.rowid,
    new Float32Array(vector)
  )
}

console.log('ðŸ‘‰ Knowledge database built')
console.log('   YOU CAN CLOSE YOUR SHELL')
