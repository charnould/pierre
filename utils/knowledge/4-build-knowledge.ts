import { $ } from 'bun'

await $`bun run utils/knowledge/1-scrap-wikipedia.ts`
await $`bun run utils/knowledge/2-generate-chunks.ts`
await $`bun run utils/knowledge/3-generate-vectors.ts`
