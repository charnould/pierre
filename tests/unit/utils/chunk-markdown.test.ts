import { expect, it } from 'bun:test'
import { split_markdown_into_chunks } from '../../../utils/knowledge/chunk-markdown'

it('should split markdown into chunks', async () => {
  const file = Bun.file('tests/unit/utils/__mocks__/markdown-file.md')
  const file_content = await file.text()

  const r = await split_markdown_into_chunks({ markdown: file_content, max_tokens: 7500 })

  expect(r.length).toBe(2)
  expect(r[0]).toMatchSnapshot()
  expect(r[1]).toMatchSnapshot()
})
