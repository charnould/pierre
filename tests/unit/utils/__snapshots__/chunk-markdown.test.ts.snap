// Bun Snapshot v1, https://bun.sh/docs/test/snapshots

exports[`should split markdown into chunks 1`] = `
"# Garantie Visale

## Qu'est-ce que la Garantie Visale ?

### Objet et objectifs

La garantie VISALE (Visa pour le Logement et l’Emploi) est une caution locative proposée par Action Logement. Elle assure au propriétaire le paiement du loyer et des charges locatives en cas de défaillance de paiement par le locataire. Cette garantie est gratuite pour le locataire ainsi que pour le bailleur ou propriétaire, et elle a pour objectif de faciliter l’accès au logement tout en offrant une sécurité supplémentaire au bailleur.

### Que prend en charge la garantie Visale ?

Visale protège les bailleurs contre deux principaux risques : les impayés et les dégradations locatives. En outre, elle prend en charge toutes les procédures de recouvrement à l’encontre du locataire.

Concernant les impayés, Visale couvre les loyers et toutes les charges récupérables dues par le locataire, ainsi que les taxes locatives inscrites au bail, les acomptes quittancés, et la régularisation annuelle des charges. Cette garantie est valable pendant toute la durée de l’occupation du logement par le locataire et couvre jusqu’à 36 mois d’impayés. Par exemple, si le locataire cesse de payer son loyer pendant 3 mois après 4 ans d’occupation, Action Logement remboursera les 3 mois de loyer impayés.

En cas de dégradations locatives, Visale couvre l’équivalent de deux mois de loyer, charges comprises, à la sortie du locataire. Action Logement se charge également des démarches de recouvrement, qu’elles soient amiables ou contentieuses.

Il est important de noter que Visale ne couvre pas le dépôt de garantie. En tant que « caution », son rôle est de fournir une assurance au bailleur en cas d’impayés, similaire à celle d’un garant solidaire (ou caution personne physique), mais c’est Action Logement qui joue ce rôle, offrant ainsi une sécurité renforcée pour le bailleur.

Enfin, Visale ne couvre pas la prime d’assurance habitation, qui est une charge récupérable auprès du locataire. Visale se limite aux charges récupérables spécifiées à l’article 7 (a) de la loi n° 89-462 du 6 juillet 1989 et au décret 87-713 du 26 août 1987.

### Combien coûte la garantie Visale ?

Que vous soyez propriétaire ou locataire, Visale est entièrement gratuit. Ce dispositif est financé par les cotisations des entreprises du secteur privé versées à Action Logement, dans le but d'aider les salariés et futurs salariés à trouver un logement.

Il est important de noter que le bailleur ne peut souscrire à une autre garantie pour les mêmes risques couverts par le dispositif Visale (comme une caution personne physique, caution personne morale, ou assurance) pendant la même période, sous peine de nullité du contrat.

### La garantie Visale est-elle cumulable avec d’autres aides Action Logement ?

Action Logement propose plusieurs aides pour faciliter l’installation dans un logement et soutenir la reprise ou l’accès au premier emploi. Ces aides sont cumulables avec Visale, sous réserve de certaines conditions :

- [L’Avance LOCA-PASS](https://www.actionlogement.fr/l-avance-loca-pass) : Cette aide permet de verser immédiatement le dépôt de garantie demandé par le bailleur et de le rembourser progressivement, sans intérêts, sur une période maximale de 25 mois. Le montant maximum de cette avance est de 1200 €. Plus d’informations sont disponibles sur Action Logement.
- [L’aide MOBILI-JEUNE](https://www.actionlogement.fr/l-aide-mobili-jeune) : Cette subvention aide à réduire le montant des quittances de loyer. L’aide varie de 10 € à 100 € maximum par mois. Pour en savoir plus, consultez Action Logement.
- Plateforme [Alternant d'Action Logement](https://alternant.actionlogement.fr): Cette plateforme fournit des informations sur les aides disponibles pour améliorer la qualité de vie des alternants pendant leur formation.

Toutes les solutions proposées par Action Logement sont accessibles via le site https://www.actionlogement.fr.

## A quels publics jeunes s’adresse Visale ?

Visale s'adresse aux jeunes de 18 à 30 ans (inclus) entrant dans un logement du parc locatif privé, indépendamment de leur situation.

Pour les étudiants et les alternants, Visale est éligible quel que soit le type de logement, qu'il soit privé, social, conventionné ou non.

De plus, Visale est disponible pour tous les jeunes de 30 ans ou moins, y compris les étudiants et les alternants, dans des structures collectives telles que les résidences universitaires, les logements foyers, et les foyers pour jeunes travailleurs.

Précisions importantes :

- Un demandeur non émancipé, âgé d'au moins 17 ans et 10 mois, peut faire une demande de visa dans le cadre du dispositif Visale. Toutefois, il ne pourra signer son bail qu'une fois majeur, conformément à la législation.
- Un locataire ou un salarié hébergé chez l'habitant est éligible, à condition qu'un contrat de bail ou une convention d'hébergement soit signé.
- Les locations dans des résidences touristiques ou dans des résidences hôtelières à vocation sociale ne sont pas couvertes par Visale.

#### Qui peut bénéficier de la garantie Visale ?

Un candidat locataire est éligible à Visale s'il remplit au moins une de ces conditions :

- J’ai entre 18 et 30 ans
- Je suis salarié de plus de 30 ans :
  - Embauché(e) depuis moins de 6 mois (hors CDI confirmé)
  - OU Gagnant jusqu’à 1500 € nets/mois
  - OU En mobilité professionnelle
  - OU En possession d’une promesse d’embauche de moins de 3 mois
- Je suis éligible au bail mobilité.
- Je suis logé par un organisme d’intermédiation locative

Le logement :

- Peut être meublé ou non
- Doit être la résidence principale du candidat locataire
- Doit avoir un loyer (charges comprises) de 1500 € maximum en Île-de-France ou de 1300 € maximum sur le reste du territoire
- Doit être décent et respecter le Règlement Sanitaire Départemental (RSD)
- Doit faire l’objet d’un bail, ou contrat de location
- Doit être situé sur le territoire français métropolitain ou les départements et régions d’Outre-Mer (La Réunion, Guyane, Martinique, Guadeloupe, Mayotte). Visale ne peut garantir les logements situés dans les COM (Collectivités d’Outre-Mer) telles Saint Barthelemy, Saint Martin, Polynésie Française, Saint Pierre et Miquelon, Wallis et Futuna, Nouvelle Calédonie.

##### Conditions d’éligibilité relatives à l’état civil du demandeur

Le dispositif Visale s’adresse aux personnes :

- De nationalité française ou résidant dans l’UE, EEE ou Suisse, Andorre, Monaco et Saint Marin ;
- Disposant d’un titre de séjour Résident ou visa en cours de validité valant autorisation de travail pour les résidents hors UE/EEE ;
- Etudiant de l’Union Européenne justifiant d’une carte d’étudiant et d’un passeport en cours de validité ;
- Etudiant hors Union Européenne justifiant d’un visa long séjour valant titre de séjour mention étudiant ou passeport talent en cours de validité, ou d’un titre de séjour mention étudiant en cours de validité (le Visa avec mention « dispense temporaire de carte de séjour » où la mention « Etudiant » n’apparait pas, n’est pas recevable pour une demande visale).

##### Conditions d’éligibilité des jeunes salariés

Aucune condition liée au contrat de travail (type ou durée), à l’employeur ou au secteur d’activité n’est requise, y compris pour un salarié employé à domicile par un particulier (exemple : assistante maternelle ou autre).

##### Conditions d’éligibilité des étudiants

Pour être éligible, un étudiant doit justifier de son inscription dans un établissement post-secondaire pour l’année scolaire en cours en France. Il doit obligatoirement fournir un justificatif d’identité en cours de validité.

##### Conditions d'éligibilité des jeunes stagiaires

Les stagiaires, de 18 à 30 ans inclus, sont éligibles à Visale. Il conviendra de fournir la convention de stage et la dernière attestation de compensation de stage. En l’absence d’attestation, l’indemnité de stage retenue est celle indiquée sur la convention.

##### Conditions d'éligibilité des jeunes salariés d’une entreprise du secteur privé ou du régime social agricole

Un jeune salarié d’une entreprise du secteur privé ou relevant du régime social agricole est éligible à Visale.

##### Conditions d'éligibilité des jeunes sans emploi ou bénéficiaires des minima sociaux

Tous les jeunes de 18 à 30 ans inclus sont éligibles à Visale y compris les demandeurs d’emploi et les bénéficiaires des minimas sociaux. Il conviendra de fournir la dernière attestation de paiement Pôle emploi (ou la notification d’ouverture de droits) ou la dernière attestation de paiement RSA (ou notification d’ouverture de droits).

##### Conditions d'éligibilité des travailleurs non salariés, indépendants ou entrepreneurs, de moins de 31 ans

Si vous avez moins de 31 ans et que vous êtes travailleur non salarié, indépendant ou entrepreneur, vous êtes éligible à Visale.

##### Conditions d'éligibilité des intermittents du spectacle de moins de 31 ans

Les intermittents de moins de 31 ans sont éligibles au dispositif Visale, quel que soit leur situation au moment de leur demande. Cependant pour calculer un loyer minimum ils doivent renseigner et justifier les revenus obtenus au cours du dernier mois.

Pour les contrats inférieurs ou égaux à un mois, l’ancienneté des justificatifs fournis doit concerner la période des 3 derniers mois précédant la demande de Visa. Il est nécessaire d’indiquer la moyenne des ressources sur les 3 mois précédant la demande.

## Quelles sont les conditions de solvabilité et de justification requises pour le locataire ?

Le ménage locataire est éligible à la garantie Visale dans la limite d’un loyer maximum, calculé à partir d’un effort logement basé sur l’ensemble des ressources des co-titulaires du bail.

Le loyer maximum est égal à 50 % de la somme de toutes les ressources des futurs co-titulaires du bail, dans la limite du plafond d’exclusion défini (1 500€ Région Île-de-France / 1 300€ pour le reste de la France).

Exemple :

- Revenus nets mensuels = 1200€ ;
- Le montant du loyer charges maximum couvert est de 600€.

Les étudiants sont éligibles sans justification de ressources dans la limite d’un loyer forfaitaire de 800 € en Région Île-de-France et de 600 € pour le reste de la France. Un étudiant peut augmenter son loyer plafond si celui-ci justifie de ressources mensuelles supérieures à 1600 € (Île-de-France) ou 1200 € (Hors Île-de-France).

Pour rappel, la liste des départements d’Île-de-France : Paris (75), Seine-et-Marne (77), Yvelines (78), Essonne (91), Hauts-de-Seine (92), Seine-Saint-Denis (93), Val-de-Marne (94), Val-d’Oise (95).

### Quelles sont les ressources prises et non prises en compte ?

Communiquer vos ressources permet de connaître le loyer maximum pour lequel Visale se portera garant.

#### Les ressources prises en compte

L’ensemble des ressources du ménage sont prises en compte :

- Revenus d’activité
- Revenus de remplacement (ex. allocations Pôle Emploi, indemnités journalières)
- Revenus dits de complément (ex. allocation logement, allocations familiales, complément familial, allocation d’éducation de l’enfant handicapé (AEEH), allocation de soutien familial, allocation journalière de présence parentale (AJPP)).

La garantie jeune (accompagnemnt qui s’adresse aux jeunes de 16 ans à moins de 26 ans, en situation de précarité qui ne sont ni en emploi, ni en formation, ni en étude) est acceptée dans les ressources prises en compte par Visale. Les pièces justificatives à fournir sont : la dernière attestation de paiement ou la copie du contrat signé avec la mission locale.

#### Les ressources non prises en compte

Les revenus de complément suivants :

- Prestations familiales (Prestation d’accueil du jeune enfant (PAJE), y compris Complément Libre Choix d’Activité) ; Allocation de rentrée scolaire.
- Prime mobilité Pôle Emploi.
- Allocation pour demandeur d’Asile (ADA).
- Les revenus exceptionnels non récurrents

Les autres revenus suivants :

- Revenus du patrimoine
- Loyers et charges perçus
- Dividendes
- Intérêts et placements
- Plus-values mobilières ou immobilières.

### Quels justificatifs doivent être fournis ?

#### Je suis un travailleur non salarié, indépendant ou entrepreneur

- Pour les auto-entrepreneurs : la déclaration mensuelle ou trimestrielle de chiffre d’affaires (faites à l’URSSAF), à défaut les autres ressources perçues par ailleurs.
- Pour les artisans, professions libérales, commerçants, expert-comptable : La Déclaration Sociale des Indépendants (DSI)
- Pour les professionnels de santé, conventionné ou non : L’avis d’imposition ou les déclarations trimestrielles URSSAF, à défaut les autres ressources perçues par ailleurs.

#### Je suis un étudiant boursier qui perçoit des revenus supérieurs au plafond forfaitaire (supérieur à 1600 € en Région Île-de-France ou supérieur à 1200 € ailleurs)

Les étudiants sont éligibles sans justification de ressources dans la limite d’un loyer forfaitaire de 800 € en Région Île-de-France et de 600 € pour le reste de la France. Un étudiant boursier ou salarié peut augmenter son loyer plafond en fournissant une notification définitive ou provisoire de bourse d’état ou les justificatifs relatifs à ses revenus (contrat de travail, bulletins de salaire…).

#### Je suis un alternant qui perçoit des revenus supérieurs au plafond forfaitaire (supérieur à 1600 € en Région Île-de-France ou supérieur à 1200 € ailleurs)

Une copie de son contrat d’apprentissage ou de professionnalisation et le bulletin de salaire du mois de la demande de visa.

## Comment obtenir la caution Visale ?

### Étape 1 sur 2 : Obtention d'un Visa pour le futur Locataire

#### Qu'est-ce qu'un visa Visale ?

Un visa Visale est une attestation qui certifie que le locataire potentiel bénéficie de la garantie Visale, proposée par Action Logement, qui s'engage à être son garant. Pour obtenir ce visa, le locataire doit créer un compte sur www.visale.fr et en faire la demande.

Ce visa est nécessaire pour activer le contrat de cautionnement Visale et doit être présenté au bailleur.

#### Processus de demande de visa

La demande de visa, initiée par le locataire, se fait en ligne sur le site www.visale.fr. Il est conseillé de la réaliser avant de commencer la recherche de logement. Une fois la demande soumise, elle est validée en quelques jours ouvrés, à condition d'être complète et conforme. Le locataire peut ensuite télécharger le visa certifié depuis son compte et le fournir au bailleur lors de la recherche de logement.

Le visa validé contient les informations suivantes :

- Nom et prénom du locataire (ou des locataires en cas de colocation)
- Montant maximal du loyer couvert (charges comprises)
- Date de validité (jusqu'à laquelle le bailleur peut activer la garantie)

Une fois la demande de visa validée, elle ne peut plus être modifiée. Si des modifications sont nécessaires, le locataire doit annuler le visa existant et soumettre une nouvelle demande, sous réserve que le visa ne soit pas lié à un contrat de cautionnement.

La validité du visa est de 3 mois maximum (6 mois pour les étudiants et alternants), calculée automatiquement en fonction de la situation du demandeur. Elle est indiquée sur le visa.

Il n'est pas possible de posséder deux visas valides simultanément. Si la période de validité du visa est dépassée, il devient obsolète, et le demandeur peut soumettre une nouvelle demande, à condition de toujours répondre aux critères d'éligibilité de Visale.

#### Utilisation du visa pour un nouveau logement

Un locataire qui a déjà bénéficié de la garantie Visale pour un logement précédent peut à nouveau en profiter pour un nouveau logement :

- en utilisant le même visa, s'il est encore valide et à condition d'être à jour dans le paiement des loyers de l'ancien logement ;
- ou en obtenant un nouveau visa si le précédent est expiré.

#### Que faire en cas de demande de visa incomplète ou non conforme ?

Si une demande de visa est signalée comme incomplète, le locataire doit vérifier les éléments manquants dans son espace personnel sur https://www.visale.fr et compléter le dossier. En cas de non-conformité, une nouvelle demande de visa doit être formulée en tenant compte des commentaires reçus, avec tous les justificatifs requis.

En cas d'incomplétude, le locataire a 15 jours pour compléter son dossier après une relance. Passé ce délai, la demande devient caduque, et il faudra reformuler une nouvelle demande.

### Etape 2 sur 2 : le bailleur active la garantie Visale

Pour activer la garantie Visale, le propriétaire doit créer un compte sur le site https://www.visale.fr, saisir le nom du locataire et le numéro de visa et signer électroniquement le contrat de cautionnement. Le bail peut ensuite être signé entre le locataire et le bailleur. En effet :

- Le contrat de cautionnement est disponible immédiatement sur l’espace personnel du bailleur dès lors que ce dernier valide sa demande de cautionnement.
- La validation en ligne du contrat de cautionnement fait office de contractualisation entre les parties. Aucun exemplaire du contrat n’est à retourner à la caution pour signature.

Par ailleurs, lors de sa demande de cautionnement sur le site www.visale.fr, le bailleur pourra vérifier la validité du visa présenté par le locataire et n’aura aucune pièce à fournir. En effet, les pièces justificatives lui seront uniquement demandées en cas de mise en jeu de la garantie suite à impayés de loyer.

Attention cependant, le bailleur, muni du visa remis par le locataire, doit formaliser sa demande de cautionnement avant (ou le jour même) la signature du bail et dans la limite de la validité du Visa du locataire. Aucune caution ne pourra être délivrée si le bail est déjà signé. Par ailleurs, la date d’effet du bail doit intervenir au plus tard au 1er du mois qui suit la date de fin de validité du visa. Exemple : Validité du visa du locataire du 01/01 au 25/02 et validation du contrat de cautionnement par le bailleur le 15/02. La date de signature du bail peut intervenir à partir du 15/02 et au plus tard le 25/02. La date d’effet du bail peut intervenir au plus tard le 01/03.

Pour information :

- Un bailleur peut bénéficier du cautionnement Visale pour plusieurs logements en même temps dès lors que les locataires sont éligibles, donc détenteurs d’un visa.
- Un bailleur peut bénéficier de Visale plusieurs fois pour le même logement, puisqu'à chaque départ du locataire, il peut souscrire un nouveau contrat de cautionnement Visale pour le nouveau locataire, dès lors que celui-ci est éligible (c’est-à-dire détenteur d’un visa certifié).
- Si la validation du contrat de cautionnement ne se concrétise pas par la signature du bail, le contrat de cautionnement est sans effet. Le bailleur peut l’annuler via son espace personnel.
- Au départ du ou des locataires en cours de bail sans déclaration d’impayé, le bailleur peut et doit actualiser son contrat depuis l’espace personnel du bailleur.
- Un bailleur a la capacité de transférer la gestion de son bien ou d’en reprendre la gestion en utilisant la fonction transfert de mon contrat qui est disponible dans son espace personnel.
- Un bailleur ne peut pas bénéficier de Visale si son bail est signé avec ses ascendants ou descendants directs (c’est-à -dire grands-parents, parents, ou enfants).

## Quelles sont les obligations du bailleur pour bénéficier de la garantie Visale ?

### Conditions du bailleur liées au loyer

Le montant du loyer (charges comprises) maximum autorisé est de 1500 euros sur le Région Île-de-France et 1300 euros sur le reste de la France. Pour rappel : les départements d’Île-de-France sont Paris (75), Seine-et-Marne (77), Yvelines (78), Essonne (91), Hauts-de-Seine (92), Seine-Saint-Denis (93), Val-de-Marne (94), Val-d’Oise (95).

### Obligations du bailleur en matière de logement décent

Qu’il soit loué vide ou meublé, le propriétaire doit fournir au locataire un logement décent. Cette obligation concerne la résidence principale du locataire. Un logement est décent si :

- la sécurité des locataires est assurée,
- la santé des locataires est préservée,
- les équipements essentiels sont fournis : coin cuisine avec évier, eau chaude et froide, installation permettant un chauffage normal…,
- il est protégé contre les infiltrations d’air parasites et permet une aération suffisante,
- il est exempt de nuisibles ou parasites.
- il est qualifié énergiquement (DPE)

Le dispositif Visale facilitant l’accès au logement du parc locatif privé s’inscrit dans la démarche de respect des règles en matière de décence de logement, seuls les logements respectant les conditions légales au titre d’un DPE conforme aux exigences édictées depuis le 1er janvier 2023 dans le cadre des lois ELAN et Climat et Résilience peuvent ainsi être couverts par Visale.

Nous vous invitons à vous renseigner sur vos obligations concernant la réalisation du diagnostic de performance énergétique via le lien suivant :

- Le site officiel de l’administration Française : https://www.service-public.fr/particuliers/vosdroits/F16096
- FAQ ADEME https://observatoire-dpe.ademe.fr/faq

Pour toutes questions ou situations concernant l’insalubrité d’un logement : contactez le numéro 0806 706 806, numéro d’appel unique (coût d’un appel local). Un conseiller de l’Agence départementale pour l’information sur le logement (ADIL) sera à votre écoute et vous expliquera les démarches à effectuer. Plus d’informations sur le site de l’ANIL : https://www.anil.org/parole-expert-logement-habitat-indigne-logement-non-decent/

## En cas de colocation, comment se passe la demande de Visa par les candidats locataires ?

### Quelles sont les conditions d’éligibilité liées à la colocation ?

En cas de multiplicité de locataires, il y a deux cas de figure :

- La colocation n’est composée que de deux colocataires (avec ou sans lien familial entre les colocataires) : un bail unique et une seule garantie Visale sont suffisants. Le locataire éligible se déclare demandeur et le colocataire co-titulaire. Dans ce cas, l’ensemble du ménage sera couvert par le dispositif. Les colocataires sont éligibles à Visale dès lors que l’un des deux co-titulaires est éligible. Les revenus des deux colocataires sont pris en compte pour calculer le loyer maximum.
- La colocation est composée de trois locataires et plus : dans ce cas chaque colocataire doit faire une demande individuelle et doit signer un bail individuel respectant les plafonds de loyers garantis par Visale.

Attention, le ménage locataire qui signe le bail doit être le même que celui mentionné dans le visa et le contrat de cautionnement. Si ce n’est pas le cas avant la signature du bail, le bailleur doit inviter le ménage locataire à effectuer une nouvelle demande de visa en adéquation avec le ou les signataires prévus dans le bail. A défaut, en cas de mise en jeu de la caution, le contrat de cautionnement sera nul.

### Que se passe-t-il en cas de départ d’un des deux co-titulaires ou colocataires ?

En cas de congé valide d’un des co-titulaires du bail, le bailleur sera toujours garanti par la caution pour le co-titulaire du bail restant. Le co-titulaire sorti restera redevable des impayés de loyer constitués avant son départ et jusqu’à la fin de la solidarité (légale ou conventionnelle de 6 mois maximum) qui le lie avec le co-titulaire restant dans le logement. Visale cesse de plein droit si un avenant est fait au bail.

## Les garanties détériorations

### Pour quels logements les dégradations locatives sont-elles prises en charge ?

Les dégradations locatives concernent les logements du parc locatif privé souscrits à partir du 01/02/19 (Il n’y a pas de rétroactivité pour les anciens contrats). Les logements du parc social ou situés en structures collectives ne sont pas couverts pour la dégradation locative.

### Quel est le montant de la prise en charge ?

Les frais de remise en état en cas de dégradations imputables au locataire sont couverts jusqu’à deux mois de loyer et charges inscrits au bail, en complément du dépôt de garantie (sauf pour le bail mobilité).

### Les dégradations locatives sont-elles prise en charge en présence d’un bail mobilité ?

Oui. Dans le cadre d’un bail mobilité, les dégradations locatives sont prises en charge à concurrence d’un montant maximum de deux mois de loyer et charges inscrits au bail.

### Quelles sont les dégradations locatives couvertes ?

Les frais de remise en état du logement en cas de dégradations locatives imputables au locataire.

### Quelles sont les dégradations locatives non couvertes ?

Les dégradations locatives non couvertes sont :

- L’usure normale, la vétusté, les dommages causés aux espaces verts, aménagements extérieurs, arbres, plantations, éléments de clôture et de fermeture de terrains privatifs.
- Les dommages causés aux biens mobiliers (meubles meublants) et aux éléments d’équipement qui peuvent être enlevés ou démontés sans les détériorer ou détériorer leur support.
- Les dommages causés par la transformation des lieux loués, autorisée par le bailleur.

### Comment demander une prise en charge en cas de dégradations locatives ?

La demande de prise en charge des dégradations locatives doit être faite sur visale.fr depuis l’espace personnel du bailleur. Le bailleur devra fournir notamment un état des dommages (état des lieux contradictoire d’entrée et de sortie, solde de tout compte transmis au locataire détaillant les dommages imputables à son occupation), ainsi que les justificatifs des dépenses de réparation.

Le bailleur s’engage à déclarer dans un délai de 30 jours sa demande de prise en charge de dégradations locatives à Visale en cas d’état des lieux de sortie mentionnant des indemnités de réparations locatives et à défaut de paiement du locataire dans les 30 jours suivant la sortie des lieux du locataire (caractérisée par la remise des clés ou l’état des lieux de sortie ou la reprise des lieux par un mandataire).

### Le bailleur peut-il cumuler une déclaration d’impayés et une détérioration locative ?

La déclaration de détérioration locative fait l’objet d’une déclaration séparée de la déclaration des impayés de loyers. Un autre numéro de dossier vous est attribué. Elle peut faire l’objet d’une déclaration seule ou à la suite d’une déclaration d’impayés de loyers au moment de la sortie du locataire.

## La gestion des impayés de loyer par le bailleur

### A partir de quel moment le bailleur doit-il déclarer un impayé de loyer ?

Dès que vous avez plus d’un mois d’impayés de loyers et charges, vous pouvez déclarer votre impayé à Visale. Autrement dit, dès que l’impayé est « constitué », vous pouvez « mettre en jeu » la garantie.

La constitution de l’impayé peut être total ou partiel :

- L’impayé de loyer total est constitué lorsque 2 mois de loyer + charges, consécutifs ou non, sont totalement impayés (aides au logement perçues par le bailleur déduites)
- L’impayé de loyer partiel est constitué si le cumul des impayés partiels, consécutifs ou non, dépasse 1 mois de loyer + charges, (aides au logement perçues par le bailleur déduites).

### Les charges seules peuvent-elles faire l’objet d’une première déclaration ?

Non, la première mise en jeu de la garantie ne peut intervenir qu’en cas de constitution d’impayé de loyer total ou partiel.

### Comment déclarer les impayés ?

Une aide à la saisie est proposée dans l’écran de déclaration. Cliquer sur le point d’interrogation situé à côté de « DECLARER MES IMPAYES ».

Un tutoriel de 4 minutes est aussi disponible sur notre site. Vous y découvrirez les étapes et démarches pour déclarer un impayé de loyer accessible en cliquant ICI.

Les pièces justificatifs nécessaires à une déclaration d’impayé sont :

- le bail
- la lettre RAR de mise en demeure
- la carte d’identité ou KBIS
- le mandat de gestion
- la taxe foncière
- le RIB

* Lors de la première déclaration d’impayés, le bailleur peut adresser la preuve de dépôt pour ne pas être hors délais. Lorsqu’il reçoit l’Accusé-Réception ou le retour du courrier « non réclamé » il doit en adresser copie via sa messagerie dans son espace bailleur.

### Quelles sont les étapes à suivre par le bailleur au moment de la survenance de l’impayé de loyer ?

A compter de la survenance de l’impayé de loyer, le bailleur s’engage à relancer le locataire par lettre simple.

A compter de la constitution de l’impayé par le locataire, le bailleur s’engage :

- Au plus tard dans les 15 jours calendaires suivant la constitution de l’impayé, à adresser au locataire par lettre RAR une mise en demeure de régler sous huitaine ;
- Dans un délai de 30 jours calendaires, à en informer la caution en déclarant l’impayé via son espace personnel.

Exemple pour un impayé de loyer total :

- Date d’exigibilité loyer N°1 : 05/07/21
- Date de survenance : 06/07/21
- Envoi lettre simple : 16/07/21
- Date d’exigibilité loyer N°2 : 05/08/21
- Date de constitution de l’impayé : 06/08/21
- Envoi mise en demeure par lettre A/R par le bailleur : 21/08/21 au plus tard
- Déclaration impayés à Action Logement : 06/09/21 au plus tard

### Quand déclarer un impayé partiel ?

La date de constitution de l’impayé de loyer partiel correspond à la date à laquelle le montant total des impayés de loyer partiels dépasse 1 mois de loyer et charges.

Exemple avec un montant de loyer de 250€ et un impayé partiel de 100€ à M, M+1 et M+2 :

- Date d’exigibilité loyer N°1 : 05/07/21
- Date de survenance N°1 : 06/07/21
- Date d’exigibilité loyer N°2 : 05/08/21
- Date de survenance N°2 : 06/08/21
- Date d’exigibilité loyer N°3 : 05/09/21
- Date de constitution de l’impayé : 06/09/21
- Envoi mise en demeure par lettre A/R par le bailleur : 21/09/21 au plus tard
- Déclaration impayés à Action Logement : 06/10/21 au plus tard

### Que doit faire le bailleur si le locataire perçoit des allocations logements ?

En cas d’impayé, le Bailleur est tenu de demander le règlement en tiers payant si le Locataire bénéficie d’une aide au logement, dès la réception de l’information par la CAF.
"
`;

exports[`should split markdown into chunks 2`] = `
"# Garantie Visale

## La gestion des impayés de loyer par le bailleur

### Quelles sont les obligations du bailleur une fois la déclaration d’impayé validée par la caution ?

Le bailleur s’engage à informer à tout moment la caution ou au plus tard aux échéances prévues au contrat, de tout évènement impactant le suivi du dossier (nouvel impayé, encaissement direct de fonds transmis de la part du locataire, départ ou abandon du logement).

Le bailleur s’engage également à actualiser la situation du locataire via son espace personnel aux échéances prévues au contrat.

### Que doit faire le bailleur (déjà indemnisé) si le locataire lui rembourse ses impayés ?

Si le cas se présente, le bailleur doit reverser le montant indemnisé par Action Logement via l’espace personnel en cliquant sur Action / Actualisation du compte locataire/ Remboursement d’un trop perçu.

### La procédure de recouvrement sera-t-elle menée par la caution ?

Action Logement prend en charge la procédure de recouvrement qui peut aller jusqu’à l’expulsion du locataire si nécessaire. Avant d’engager la procédure contentieuse, Action Logement propose à votre locataire un traitement amiable. Il peut définir son plan de remboursement directement depuis son espace locataire et régler en ligne. le locataire dispose de 2 mois pour solliciter une procédure amiable. À défaut, la procédure contentieuse se poursuit. Vous êtes informés des étapes principales de la procédure dans votre espace.

### Que faire en cas de nouvel impayé ?

Le bailleur n’a plus besoin de faire de mise en demeure au locataire, Action Logement prend en charge toute la procédure. Il doit néanmoins actualiser chaque mois le compte locataire et déclarer le loyer et les charges dus, ainsi que les éventuelles sommes perçues de la part du locataire.

Le bailleur dispose de 30 jours à compter de la date d’exigibilité du dernier loyer impayé pour faire sa déclaration.

Exemple : nouvel impayé au 05 septembre (jour d’exigibilité)

- Déclaration nouvel impayé à compter du 06 septembre et jusqu’au 5 octobre maximum.

### Quels sont les délais de paiement par la caution ?

Après la déclaration initiale d’impayé de loyer, si le dossier est complet et conforme, le bailleur reçoit une réponse sous 15 jours ouvrés avec une quittance subrogative précisant le montant qui sera versé.

Dès acceptation et validation par le bailleur de la quittance subrogative transmise sur l’espace personnel via la messagerie Visale, le remboursement des loyers est effectué par virement sur le compte bancaire du bailleur. Pour visualiser la quittance subrogative et les messages reçus : cliquer sur ACTION, Accueil, puis Messagerie.

La quittance subrogative est importante : elle permet à Action Logement de recouvrer les sommes impayées auprès du locataire. Une fois validée, le versement de l’impayé est déclenché.

### Le Locataire a quitté le logement et rendu les clés

Qu’il soit parti en respectant le préavis ou qu’il ait quitté le logement sans prévenir, le bailleur doit signaler la sortie du locataire dans son espace bailleur sur visale.fr en cliquant sur « Action ».

Les impayés de loyer doivent être déclarés avant la sortie du locataire. Si vous avez moins d’un mois d’impayé et charge, vous pourrez mobiliser le dépôt de garantie.

### Le locataire ne paie pas le dernier mois de loyer, que devez-vous faire ?

Si l’impayé n’excède pas un mois de loyer + charges, la garantie Visale ne pourra pas être mise en jeu (l’impayé ne pourra pas être déclaré dans votre espace). Le dépôt de garantie du locataire pourra alors être mobilisé.

Il vous est possible de déclarer le dernier mois de loyer en même temps que les dégradations éventuelles. En cas de dégradations locatives, Visale prendra en charge les travaux à hauteur maximum de 2 mois de loyer + charges.
"
`;
